src/reference/asciidoc/router.adoc

```diff
@@ -589,6 +589,35 @@ The example below demonstrates a `PayloadTypeRouter` configuration which is equi
 </int:payload-type-router>
 ----
 
+The equivalent router, using Java configuration:
+
+[source, java]
+----
+@ServiceActivator(inputChannel = "routingChannel")
+@Bean
+public PayloadTypeRouter router() {
+    PayloadTypeRouter router = new PayloadTypeRouter();
+    router.setChannelMapping(String.class.getName(), "stringChannel");
+    router.setChannelMapping(Integer.class.getName(), "integerChannel");
+    return router;
+}
+----
+
+and using the Java DSL:
+
+[source, java]
+----
+@Bean
+public IntegrationFlow routerFlow() {
+    return IntegrationFlows.from("routingChannel")
+            .route(new PayloadTypeRouter(), m -> m
```

Comment:
There is nice solution for raw Java on the matter:
```
.<Object, Class<?>>route(Object::getClass, m -> m
                            .channelMapping(String.class, "stringsChannel")
                            .channelMapping(Integer.class, "integersChannel"))
```
On the other hand your sample doesn't work.
Because of another method selection:
![wrong_route](https://cloud.githubusercontent.com/assets/1071590/24067160/45ec461c-0b4f-11e7-9a7b-54d3de6c8186.png)